api/token.go

```diff
@@ -10,7 +10,8 @@ type TokenService interface {
 	CheckToken(request *http.Request) (string, int, error)
 	ExtractToken(request *http.Request) string
 	CurrentAudience(request *http.Request) string
-	NewToken(id int, audience string) (string, time.Time, error)
+	SessionID(request *http.Request) string
+	NewToken(id int, sessionID, audience string) (string, time.Time, error)
```

Comment:
I generally avoid bike-shedding variable names, but if sessionID is a SAML session Index, then it might be more clear to just use `sessionIndex` uniformly, since session id is such a generic, overused term.

Reply:
Sure, I was trying to make it independent of SAML, but that made it less interesting.